home/tests/test_assessment_import_export.py

```diff
@@ -79,16 +79,17 @@ def decode_json_fields(page: DbDict) -> DbDict:
 
 def _remove_fields(pages: DbDicts, field_names: set[str]) -> DbDicts:
     for p in pages:
-        fields = {k: v for k, v in p["fields"].items() if k not in field_names}
-        yield p | {"fields": fields}
+        p["fields"] = {k: v for k, v in p["fields"].items() if k not in field_names}
+    return pages
```

Comment:
This change modifies the input, which the previous implementation didn't do. Is that guaranteed safe? Should we perhaps build a new list instead modifying the items in the one we get?

(Several other changes in this file do the same -- I won't note each case individually for now.)

Reply:
Yes I did think of this, but the way we're using it is `assessments = ff(assessments)`, so we're throwing away the old list on every application of a filter anyway, so I thought it would be fine to modify in place.